tools/devops/automation/templates/agent-pool-selector.yml

```diff
@@ -0,0 +1,65 @@
+#
+# Selects appropriate agent pool based on trigger type (PR or CI)
+#
+parameters:
+  agentPoolPR: 'VSEng-Xamarin-RedmondMacCatalinaBuildPool-iOS-Untrusted'
+  agentPoolPRUrl: 'https://devdiv.visualstudio.com/DevDiv/_settings/agentqueues?queueId=2734&view=agents'
+  agentPoolCI: 'VSEng-Xamarin-RedmondMacCatalinaBuildPool-iOS-Trusted'
+  agentPoolCIUrl: 'https://devdiv.visualstudio.com/DevDiv/_settings/agentqueues?queueId=2748&view=agents'
+  condition: succeeded()
+
+steps:
+  - powershell: |
+      $buildReason = "$(Build.Reason)"
+      $buildSourceBranchName = "$(Build.SourceBranchName)"
+      $agentPoolPR = "${{ parameters.agentPoolPR }}"
+      $agentPoolPRUrl = "${{ parameters.agentPoolPRUrl }}"
+      $agentPoolCI = "${{ parameters.agentPoolCI }}"
+      $agentPoolCIUrl = "${{ parameters.agentPoolCIUrl }}"
+
+      Write-Host "buildReason: ${buildReason}"
+      Write-Host "buildSourceBranchName: ${buildSourceBranchName}"
+      Write-Host "agentPoolPR: ${agentPoolPR}"
+      Write-Host "agentPoolPRUrl: ${agentPoolPRUrl}"
+      Write-Host "agentPoolCI: ${agentPoolCI}"
+      Write-Host "agentPoolCIUrl: ${agentPoolCIUrl}"
+
+      $agentPool = $agentPoolPR       # Default to Catalina PR pool
+      $agentPoolUrl = $agentPoolPRUrl
+      Write-Host "Default agent pool: ${agentPool}"
+
+      [bool] $isTopicBranch = $False
+      [bool] $isPullRequest = $False
+
+      if (-not ($buildSourceBranchName -eq 'main' -or $buildSourceBranchName -eq 'master' -or $buildSourceBranchName.StartsWith('d16-'))) {
+        $isTopicBranch = $True
+      }
+
+      if ($buildReason -eq 'PullRequest') {
+          $prTargetBranchName = "$(System.PullRequest.TargetBranch)"    # This system variable is only defined (and in turn the value macro replaced) when $buildReason is 'PullRequest'. Consequently, it cannot be defined as part of an input parameter
+          Write-Host "prTargetBranchName: System.PullRequest.TargetBranch: ${prTargetBranchName}"
+          $isPullRequest = $True
+          $targetBranch = $prTargetBranchName
+      } else {
+          $targetBranch = $buildSourceBranchName
+      }
+
+      Write-Host "Settings:"
+      Write-Host "  targetBranch: ${targetBranch}"
+      Write-Host "  isTopicBranch: ${isTopicBranch}"
+      Write-Host "  isPullRequest: ${isPullRequest}"
+
+      if ($isTopicBranch -or $isPullRequest) {
+        $agentPool = $agentPoolPR      # Untrusted on-prem iOS pool used for PRs targeting main and d16-x release branches
+        $agentPoolUrl = $agentPoolPRUrl
+      } else {
+        $agentPool = $agentPoolCI      # Trusted on-prem iOS pool used for CIs and release branches
+        $agentPoolUrl = $agentPoolCIUrl
```

Comment:
@mjbond-msft You are the person to answer this question.

Reply:
Tightened up the comments to describe more precisely what trigger types & branches each pool (trusted & untrusted) serves

Reply:
What's the reason for the split? According to the comments it seems to be because the `main` and `d16-*` branches are more trusted in some way, but there's no reason to trust any other (non-fork) branch more or less than `main`/ `d16-*` as far as I know.

Reply:
The intent of the split is to isolate hardware pools so that the CI builds for reviewed code (i.e., code that has completed PR review & has been merged to staging (main & release branches -- aka branches with policies associated) is handled by dedicated "trusted" hardware; whereas, all other jobs (commits to feature/topic branches) and PR merge commits are handled by an untrusted pool.  Helps better protect machines in the trusted pool that produce bits that ship in the product from being compromised. It may be worth having a discussion on what makes the most sense for the iOS team from this perspective

Reply:
So if I understand correctly, the intent is to protect the bots that produce our releases from _accidental_ damage by any of us (by creating a branch to test things and that ends up breaking the bot somehow). If that's the case, it should be explained in a comment, because my first thought was that it was trying to protect something from malicious use, and this split does not help in that scenario at all.